fence/README.md

```diff
@@ -0,0 +1,156 @@
+# Network Fencing
+
+## Terminology
+
+| Term     | Definition                                                                            |
+| -------- | ------------------------------------------------------------------------------------- |
+| CO       | Container Orchestration system that communicates with plugins using CSI service RPCs. |
+| SP       | Storage Provider, the vendor of a CSI plugin implementation.                          |
+| DR       | Disaster Recovery.                                                                    |
+| RPC      | [Remote Procedure Call](https://en.wikipedia.org/wiki/Remote_procedure_call).         |
+
+## Objective
+
+Define a standard that will enable storage vendors (SP) to
+perform node level fencing using corresponding CIDR blocks.
+
+### Goals in MVP
+
+The new extension will define a procedure that allows SP to block
+access to any client using corresponding CIDR blocks, and unblock
+as per the requirement.
+
+If multiple COs are accessing a shared resource provided by the SP, and a node
+failure takes place, this procedure will allow the SP to block access from any
+node of CO, and thus preventing further access and corruption of data.
+
+### Non-Goals in MVP
+
+* Application level fencing, i.e blocklisting by maintaining a list of applications
+  that are to be denied access.
+
+## Solution Overview
+
+This specification defines an interface along with the minimum operational and
+packaging recommendations for a storage provider (SP) to implement fencing operation.
+The interface declares the RPCs that a plugin MUST expose.
+
+Since the fencing is network based, the fencing mechanism will be most aptly implemented
+by the CSI Controller.
+
+## RPC Interface
+
+* **FenceController Service**: The Controller plugin MUST implement these sets of
+  RPCs.
+
+```protobuf
+syntax = "proto3";
+package fence;
+
+import "github.com/container-storage-interface/spec/lib/go/csi/csi.proto";
+import "google/protobuf/descriptor.proto";
+
+option go_package = ".;fence";
+
+// FenceController holds the RPC method for performing fencing operations.
+service FenceController {
+  // FenceClusterNetwork RPC call to perform a fencing operation.
+  rpc FenceClusterNetwork (FenceClusterNetworkRequest)
+  returns (FenceClusterNetworkResponse) {}
+
+  // UnfenceClusterNetwork RPC call to remove a list of CIDR blocks from the
+  // list of blocklisted/fenced clients.
+  rpc UnfenceClusterNetwork (UnfenceClusterNetworkRequest)
+  returns (UnfenceClusterNetworkResponse) {}
+
+  // ListClusterFence RPC call to provide a list of blocklisted/fenced clients.
+  rpc ListClusterFence(ListClusterFenceRequest)
+  returns (ListClusterFenceResponse){}
+}
+```
+
+### FenceClusterNetwork
+
+```protobuf
+// FenceClusterNetworkRequest contains the information needed to identify
+// the cluster so that the appropriate fencing operation can be
```

Comment:
may be good to mention it as `identify the cluster network CIDR` so that it avoid the vagueness of is it a CO cluster or storage cluster ..etc.

Reply:
Updated it to be storage cluster now :+1: